codegen/module_system.go

```diff
@@ -563,16 +563,27 @@ type ClientsInitGenerator struct {
 func (g *ClientsInitGenerator) Generate(
 	instance *ModuleInstance,
 ) (*BuildResult, error) {
-	clients := readClientDependencySpecs(instance)
+	clients := []*ClientSpec{}
+	instances := sortInstances(instance.ResolvedDependencies["client"])
+
+	for _, instance := range instances {
+		clients = append(clients, instance.GeneratedSpec().(*ClientSpec))
+	}
 
 	includedPkgs := []GoPackageImport{}
-	for _, client := range clients {
+	for i, client := range clients {
 		// TODO: there shouldn't be a special thing for custom
 		if client.ClientType == "custom" {
 			includedPkgs = append(includedPkgs, GoPackageImport{
 				PackageName: client.CustomImportPath,
 				AliasName:   client.ImportPackageAlias,
 			})
+			if len(instances[i].ResolvedDependencies["client"]) > 0 {
+				includedPkgs = append(includedPkgs, GoPackageImport{
+					PackageName: instances[i].PackageInfo.GeneratedPackagePath,
+					AliasName:   instances[i].PackageInfo.GeneratedPackageAlias,
+				})
+			}
 			continue
```

Comment:
You probably want to avoid continues over use of if/else. Flow control is generally more easily followed with if structures than through continue/return statements.